### ISSUE TITLE
Project Feedback/Reviews

### ISSUE THREAD

**Issue by darkstego** [darkstego/Mudeer#8]
I couldn't find a feedback section for Github projects so I am keeping it as an issue. If you have time and have used the project please reply below with any feedback about the project. What are things you like or don't like? Are there special ways you use the project. This isn't the place for specific bugs or feature requests. But it is very useful to know what people find good and what they don't.

**Comment by pzy** [comment]
Works great for me, thanks for that.
Only thing i would 'complain' is that after installation Meta+A worked for me but S,D not, i expected all of the example shortcuts to work after installation.

**Comment by darkstego** [comment]
> Only thing i would 'complain' is that after installation Meta+A worked for me but S,D not, i expected all of the example shortcuts to work after installation.

I did try to fix that. But as far as I know there is no way for a script to overwrite existing shortcuts and if there are any collisions the shortcuts need to be manually setup. Not the best install experience but I don't know if there is any way to improve it currently.

**Comment by insspb** [comment]
Works fine in 22 04. Thank you.

**Comment by upnix** [comment]
Running Kubuntu 22.10 (Plasma 5.25.5), working great for me. Your KDE Store instructions are accurate, but somehow I missed them and had to work it out myself. My one suggestion: You should provide a way people can "thank you" with donations.

Having tried to make Plasma-supplied windowing shortcuts work for me for a while, I had lots of shortcuts that overlapped with Mudeer's defaults. The number of potential window placements is overwhelming and there isn't a diagram to help reason about them all. What I eventually did was remove all the Mudeer shortcuts and come up with my own system.

![image](https://user-images.githubusercontent.com/6163553/218138246-114d1eee-fe4f-4d86-b4af-1f8dc3676955.png)

Rather than have a shortcut for every possible position, I limit the number of positions Mudeer has shortcuts for, then use KDE's "Move window Up/Down/Left/Right" and "center window" shortcuts to move the correctly sized window to where I want it. This greatly reduces the number of shortcuts that I have to remember, and that might conflict with other shortcuts.

So in my setup, to get a quarter-sized window in the bottom left I hit:

1. Meta+Shift+Alt+1 (Bottom, center left)
2. Meta+Shift+Left arrow (Move to far left)

The first isn't really easy to make work with one hand, so you could just always have "top quarter," "left quarter," then use two window moving shortcuts.

Anyway, not a suggestion really, just sharing what I've done. Thank you for the great script!

**Comment by Ganneff** [comment]
> If you have time and have used the project please reply below with any feedback about the project. What are things you like or don't like? Are there special ways you use the project. This isn't the place for specific bugs or feature requests. But it is very useful to know what people find good and what they don't.

Is there a way to get you some donation?

**Comment by sandro-mancuso** [comment]
Hi,

I've found your kwin script to be great, and even on 5.27's new tiling funcitonality (which is awesome too!), I like yours better because that one's still showing its growing pains!  I think the fullscreen in a tile, and the shortcuts are big benefits.

One thing I'd like, is to be able to have an overlay of both 3-row and 2-row positioning options.    The goal would be a layout like the two in the image below:

![image](https://user-images.githubusercontent.com/4250785/224595283-f8b48c46-d61b-46ef-89d0-149d95f85ddf.png)

While working, I find this useful for the way I work, so that I can keep  certain windows opened, and away to the sides, small, but still visible, and then have focus on the main window.  A layout of this nature also helps keep key windows up near the top so that when in a video call, if I'm concentrating on what's on the screen, they don't just see the top of my head ;) .  Some windows I would stack in three, others just two.

Some nice-to-have ideas, just in case you take a liking to one and want to take a crack at it:

Combined with the layouts I drew above, if I could designate an "active" tile (that large one in the middle in my example), make it such that I can alt-tab to change the active app, and have it automatically be placed in the "active" tile,  and swapping the previously active app into it's old location

have a similar override to the width of the middle half, similar to what you've done with thirds, in the script settings.  

I'm using window rules to more-or-less do this now, but you may like the idea of integrating into your tool the ability to assign window classes or apps to always snap to a particular grid location.

While I'm spewing out wishlists, one last one:  Remember the window size/location so that I can effectively issue a shortcut to snap a window into a tile, and then use the same shortcut - effectively as a toggle - that would undo that and put the window back where it was.

**Comment by jp-bennett** [comment]
This project is great, and really makes a big 16:9 4k monitor more usable. I pretty exclusively use it to split the big monitor into 4 quarters, and putting an app in each corner. It even works to push a full-screen app to one-quarter size, which is great.

The only annoyance I've run into isn't even Mudeer's fault, but it might be addressable here. When Chrome is running in a corner of the screen as described, occasionally clicking on a tab will move the entire Chrome window out of place. It would be helpful if it were possible to lock/pin a window into a specific location, so that no matter how you clicked on it, it stays put.

For bonus points, this could also apply when using the application's full-screen function. If Chrome is locked to the lower left corner, then going full-screen in a Youtube video should do the fullscreen tiling trick.

**Comment by darkstego** [comment]
> Is there a way to get you some donation?

Thanks for the offer, but I have no way of accepting donations at this time. But I do greatly appreciate the positive feedback.

**Comment by darkstego** [comment]
@jp-bennett, glad the script works well for you.

>The only annoyance I've run into isn't even Mudeer's fault, but it might be addressable here. When Chrome is running in a corner of the screen as described, occasionally clicking on a tab will move the entire Chrome window out of place. It would be helpful if it were possible to lock/pin a window into a specific location, so that no matter how you clicked on it, it stays put.

I do occasionally come across similar issues. Or there is the case of creating a new Chrome window and it would be ever so slightly out of position from the previous tiling point. It would be nice to be able to snap windows that are very close to the tiling spot to the actual tiling spot. The only problem is that there are no predefined tilling spots in this project, as it is supposed to be flexible in how to tile or whether to tile at all. I guess it would be possible to keep a size and position of every possible tiling and if there is a close enough match to one then snap there. 

> For bonus points, this could also apply when using the application's full-screen function. If Chrome is locked to the lower left corner, then going full-screen in a Youtube video should do the fullscreen tiling trick.

I might add that as a feature request, to snap fullscreen to current tilling position.

**Comment by sandro-mancuso** [comment]
>That does seem interesting, but I would need to decide on how to handle https://github.com/darkstego/Mudeer/issues/10 first, because if I will be using sequential button presses, then I will no longer be able to use the shortcuts as toggles.

To be completely honest, I love your idea in #10 as a toggling use-case way, way, more than mine.  It's not that big of a pain, the whole resizing stuff, it was just an additional thought I had.  Besides, after further consideration, this represents one of the few times I'm using the mouse for window management.  If anything, the value isn't around using shortcut toggles, but to  remember the window size prior to being snapped into a grid position, and if moved out, to revert to previous size.

But to be clear, this isn't a super high value use case (to me).  I just added it in case it turned out to be easy to do.   I'm really sold on the toggle idea to walk through window placements (#10).  I like it more than any of the ideas I shared actually.
